File: esbmc.py/Testes02/validador_categoria/validador_caracteres_especiais.cpp
#include <cctype>

#define MAX_LEN 31

using namespace std;

bool validate_no_special_characters(const char category[MAX_LEN], int len) {
    const char specials[] = "!@#$%^&*()_+=-[]{}|;:,.<>?/~`";
    for (int i = 0; i < len; i++) {
        for (int j = 0; specials[j] != '\0'; j++) {
            __ESBMC_assert(category[i] != specials[j], "A categoria não deve conter caracteres especiais.");
        }
    }
    return true;
}

int main() {
    char category[MAX_LEN];
    int len;

    __ESBMC_assume(len >= 3 && len <= 30);

    validate_no_special_characters(category, len);

    return 0;
}

File: esbmc.py/Testes02/validador_categoria/validador_caracteres_especiais.py
MAX_LEN: int = 31

def validate_no_special_characters(category: str, length: int) -> bool:
    specials: str = "!@#$%^&*()_+=-[]{}|;:,.<>?/~`"
    i: int = 0
    while i < length:
        j: int = 0
        while j < len(specials):
            assert category[i] != specials[j], \
                "A categoria não deve conter caracteres especiais."
            j += 1
        i += 1
    return True


def main() -> None:
    category: str = input()    
    length: int = len(category)
    
    assert length >= 3 and length <= 30

    validate_no_special_characters(category, length)


main()
